packages/compass-web/polyfills/@mongodb-js/devtools-connect/index.ts

```diff
@@ -16,23 +20,21 @@ export async function connectMongoClient(
   delete options.parentState;
   delete options.parentHandle;
   options.__skipPingOnConnect = true;
-  const client = new MongoClient(url, options);
-  await client.connect();
+  const { client } = await devtools_connect.connectMongoClient(
+    url,
+    options,
+    logger,
+    MongoClient
+  );
   return {
     client,
     state: {
-      getStateShareServer() {
-        return Promise.resolve('Not Available');
-      },
-      oidcPlugin: {
-        logger,
-        serialize() {
-          return Promise.resolve(undefined);
-        },
-      },
-      destroy() {
-        return Promise.resolve();
+      // eslint-disable-next-line @typescript-eslint/require-await
```

Comment:
What is the motive for preferring synchronous functions that return a Promise vs using the keyword `async` that produces the same thing more succinctly? 

Happy to put this back to `Promise.resove/reject`

Reply:
> This function is async because not returning a Promise would break the API contract

Basically this trap is a little too easy to fall into esp with functions that are just for throwing. The `systemCA` function in the other file was synchronously throwing its real counter part is an async function. 

In this case, making this misstep is more about missing TS, but generally throwing the keyword around is a way to not think about what you do inside the scope of the function but always get back the return type you want (a promise). 

---

Anyway with the refactors made I think I removed any usage of this eslint exception. Fun fact `async () => Promise.resolve(...)` doesn't yell at you to await something 😅 best of both worlds??